src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -60,26 +60,31 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 	}
 
 	public async $openConnectionDialog(providers: string[], initialConnectionProfile?: IConnectionProfile, connectionCompletionOptions?: azdata.IConnectionCompletionOptions): Promise<azdata.connection.Connection> {
+		let connectionType = ConnectionType.editor;
+		if (connectionCompletionOptions && connectionCompletionOptions.saveConnection !== undefined) {
+			connectionType = connectionCompletionOptions.saveConnection ? ConnectionType.editor : ConnectionType.extension;
+		}
```

Comment:
I think one big thing I'd like to fix before merging this in is figuring out a better name to give this or to move it into a separate property. It looks like we currently use this primarily (only?) in connectionDialogService.ts and while 2 of the uses are just for whether we save the connection there's an annoying one that uses it for handling connection cancel calls. It'd be good to dig into this a little more and see if we can clean it up before merging this in.

Reply:
I've changed the name from `extension` to `temporary` to show that the connection is not going to be saved.